starknet_devnet/general_config.py

```diff
@@ -15,23 +16,38 @@
 from .constants import SUPPORTED_TX_VERSION
 from .fee_token import FeeToken
 
-DEFAULT_GENERAL_CONFIG = build_general_config(
-    {
-        "cairo_resource_fee_weights": {
-            "n_steps": constants.N_STEPS_FEE_WEIGHT,
-        },
-        "contract_storage_commitment_tree_height": constants.CONTRACT_STATES_COMMITMENT_TREE_HEIGHT,
-        "event_commitment_tree_height": constants.EVENT_COMMITMENT_TREE_HEIGHT,
-        "global_state_commitment_tree_height": constants.CONTRACT_ADDRESS_BITS,
-        "invoke_tx_max_n_steps": DEFAULT_MAX_STEPS,
-        "min_gas_price": DEFAULT_GAS_PRICE,
-        "sequencer_address": hex(DEFAULT_SEQUENCER_ADDRESS),
-        "starknet_os_config": {
-            "chain_id": DEFAULT_CHAIN_ID.name,
-            "fee_token_address": hex(FeeToken.ADDRESS),
-        },
-        "tx_version": SUPPORTED_TX_VERSION,
-        "tx_commitment_tree_height": constants.TRANSACTION_COMMITMENT_TREE_HEIGHT,
-        "validate_max_n_steps": DEFAULT_VALIDATE_MAX_STEPS,
-    }
-)
+
+class StarknetChainId(Enum):
+    """Chain id enum used for mapping."""
+
+    MAINNET = from_bytes(b"SN_MAIN")
+    TESTNET = from_bytes(b"SN_GOERLI")
+
+
+def build_config(chain_id=""):
```

Comment:
We are making a more specific function, but the name doesn't reflect that, the name is even more general that the function we are wrapping (`build_general_config`). I would rename this to e.g. `build_devnet_general_config`.

Reply:
https://github.com/Shard-Labs/starknet-devnet/pull/358/commits/19e0110dbae7c23c694b4bd006a640c6525894e8